feat(curriculum): dice game project (#50819)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/64994c4e6255ac241c0436e9.md

---
id: 64994c4e6255ac241c0436e9
title: Step 1
challengeType: 0
dashedName: step-1
---

# --description--

> In this project, you will be building a dice game. The game is simple. The player gets 6 rounds where in each round, you can roll the dice for 3 times and collect the score. 

Reviewer: ```suggestion
In this project, you will be building a dice game. The game is simple. The player gets 6 rounds where in each round, you can roll the dice for 3 times and collect the score. 
```

Author: done

---
id: 64994c4e6255ac241c0436e9
title: Step 1
challengeType: 0
dashedName: step-1
---

# --description--

In this project, you will be building a dice game. The game is simple. The player gets 6 rounds where in each round, you can roll the dice for 3 times and collect the score. 

> The very first thing you will be doing is to set the initial values for the game. First, declare variables named `rolls`, `score` and `totalScore` and set all of their values to `0`. Then, declare a variable named `round` and set its value to `1`.

Reviewer 2: ```suggestion
The HTML and CSS have already been provided for this project, feel free to have a look.

The first thing you will do is set the initial values for the game. First, declare variables named `rolls`, `score`, and `totalScore` and set all of their values to `0`. Then, declare a variable named `round` and set its value to `1`.
```

Author: done

---
id: 64994c4e6255ac241c0436e9
title: Step 1
challengeType: 0
dashedName: step-1
---

# --description--

In this project, you will be building a dice game. The player gets 6 rounds where in each round, you can roll the dice for 3 times and collect the score.

The HTML and CSS have already been provided for this project, feel free to have a look.

> The first thing you will do is set the initial values for the game. First, declare variables named `rolls`, `score`, and `totalScore` and set all of their values to `0`. Then, declare a variable named `round` and set its value to `1`.

Reviewer: ```suggestion
The first thing you will do is set the initial values for the game. First, declare variables named `rolls`, `score`, and `totalScore` and set all of their values to `0`. Then, declare a variable named `round` and set its value to `1`. These variables will change throughout the code, so use `let` when declaring all of them.
```

# --hints--

> You should declare a variable named `round`.

Reviewer: ```suggestion
You should use `let` to declare a variable named `round`.
```

```js
assert.match(code, /let\s*round/);
```

You should set the value of `round` to `1`.

```js
assert.match(code, /round\s*=\s*1/);
```

> You should declare a variable named `rolls`.

Reviewer: ```suggestion
You should use `let` to declare a variable named `rolls`.
```

```js
assert.match(code, /let\s*rolls/);
```

You should set the value of `rolls` to `0`.

```js
assert.match(code, /rolls\s*=\s*0/);
```

> You should declare a variable named `score`.

Reviewer: ```suggestion
You should use `let` to declare a variable named `score`.
```

```js
assert.match(code, /let\s*score/);
```

You should set the value of `score` to `0`.

```js
assert.match(code, /score\s*=\s*0/);
```

> You should declare a variable named `totalScore`.

Reviewer: ```suggestion
You should use `let` to declare a variable named `totalScore`.
```
